lib/all_futures/base.rb

```diff
@@ -3,26 +3,28 @@
 module AllFutures
   class Base < ActiveEntity::Base
     prepend ::AllFutures::Callbacks
-    include ActiveModel::Conversion
+    include ::ActiveModel::Conversion
+    include ::AllFutures::Persist
+    include ::AllFutures::Dirty
 
     def initialize(attributes = {})
-      super do
-        extend(::AllFutures::Persist)
```

Comment:
so as far as Im aware you cant extends / or include in an initialize method, it has to be done at the class method level. If you wanted to do this, youd have to do something like:

```rb
self.class.class_eval do
  extend ::AllFutures::Persist
end
```

which is kind gross in an initializer

Reply:
I must have created Frankenstein's monster, because the uncanny thing is that it actually *appeared* to work.

However, there's no question that it's not idiomatic Ruby and god knows what other effects doing this might have.

Reply:
Yeah, it did work in the sense that the methods were called, it just worked _too well_ as they took precedence over the prepended module methods. At some point I should read up on why `extend` can be called on an instance and what that actually does.

Reply:
Apparently extend _just works_ in initializers but include won't.

https://rubymonk.com/learning/books/4-ruby-primer-ascent/chapters/48-advanced-modules/lessons/117-included-and-extend#solution4533

Reply:
I'd declare it an anti-pattern as it stomps out anything you might have prepended.

In an adjacent but interesting development, I just pushed an update that fixed a bug preventing bang-methods from working. Essentially, https://github.com/leastbad/all_futures/blob/8ef25a689ad226a51ab32e90fd6e3dc72732b2c6/lib/all_futures/callbacks.rb#L49 could not see the `destroy!` method in this module because I was using `method_alias`. When you pass it to the Ruby `method` function, it processes it as `destroy` because an alias is not a method but a pointer to a method. And here I assumed that it was a convenience macro, like `attr_accessor`! NOPE